tests/www/test_views.py

```diff
@@ -446,5 +448,88 @@ def test_mount(self):
         self.assertIn(b"DAGs", resp_html)
 
 
+class TestTaskInstanceModelView(unittest.TestCase):
+    DAG_ID = 'dag_for_testing_setting_subdag_task_instance_state_view'
+    SUBDAG_ID = 'task_for_testing_setting_subdag_task_instance_state_view'
+    SUBDAG_NAME = '{dag_id}.{subdag_id}'.format(dag_id=DAG_ID, subdag_id=SUBDAG_ID)
```

Comment:
I'm not too familiar with subdags (as we don't use them) but is simply including a '.' what makes this a subdag? Do subdags always need a dot in the name, or is this just your convention.

i.e. is the problem with escaped characters and nothing strictly to do with subdags?

Reply:
The SubDagOperator requires a dot in their name otherwise an exception is raised. https://github.com/apache/incubator-airflow/blob/master/airflow/operators/subdag_operator.py#L53-L58
I think it's a bit odd to enforce this so heavily, but it was reviewed and agreed on given that it is in the code base. So given that the name is required to follow that format (and to continue backwards compatibility for existing SubDagOperator's), the problem is that airflow requires the escape character in these task instances and flask_admin is duplicates that escape character when trying to perform the "actions" available in the UI.

Reply:
What if I call my task "fetch_from_google.com"? Would that cause similar errors elsewhere in the web UI?

(maybe if that is a bigger change it can be done as a separate PR)

Reply:
It would cause the same problem that I've identified here within the Task Instance flask_admin view but should be caught and fixed with this line here: https://github.com/apache/incubator-airflow/pull/2691/files#diff-948e87b4f8f644b3ad8c7950958df033R2602

As to its impact across the UI, I haven't looked enough into all of it to really be able to answer that question confidently. It's something I could look at later on.